atc/creds/vault/manager.go

```diff
@@ -123,6 +130,13 @@ func (manager VaultManager) Validate() error {
 		return fmt.Errorf("path prefix must be a non-empty string")
 	}
 
+	for i, tmpl := range manager.LookupTemplates {
+		name := fmt.Sprintf("lookup-template-%d", i)
```

Comment:
This `name` argument seems a little awkward to use - is it required, or can it just be a made-up string in `creds.BuildSecretTemplate`? 🤔

Reply:
I think the only thing the names are good for, apart from maybe appearing in error messages, is for being invoked by name. Hopefully nobody is doing such a thing in the case of any of these templates! Errors are like `template: foo:1: unexpected unclosed action in command` for a template named `foo` that's defined as `"{{"`. So there's a bit of convenience value in having that bubble up with a name attached. We could create with an empty name, which yields the same sort of error message in the end, just with `template: :1:...`.